apps/meteor/server/models/raw/ModerationReports.ts

```diff
@@ -18,12 +19,11 @@ export class ModerationReportsRaw extends BaseRaw<IModerationReport> implements
 
 	modelIndexes(): IndexDescription[] | undefined {
 		return [
-			{ key: { 'ts': 1, 'reports.ts': 1 } },
-			{ key: { 'message.u._id': 1, 'ts': 1 } },
-			{ key: { 'reportedUser._id': 1, 'ts': 1 } },
-			{ key: { 'message.rid': 1, 'ts': 1 } },
+			{ key: { '_hidden': 1, 'ts': 1, 'reports.ts': 1 } },
 			{ key: { userId: 1, ts: 1 } },
 			{ key: { 'message._id': 1, 'ts': 1 } },
+			{ key: { '_hidden': 1, 'message.u._id': 1, 'ts': 1 } },
```

Comment:
I'm not totally sure the queries `{ _hidden: { $ne: true } }` will actually use this indexes.. can you please confirm by by doing an `.explain('executionStats')` on mongoshell?

it is also clear  that all indexes share the same `ts` field, I wonder if we can optimize those indexes by having less compound indexes being used by multiple queries.

Reply:
Thank you, yes, it uses these indexes. I could not find a better solution for optimizing the common `ts` field because of respecting the ESR Rule. 
Both common fields `_hidden` and `ts` use range operators.